jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -5673,9 +5673,36 @@ private static ReflectionFactory getReflectionFactory() {
 	/*[ENDIF] JAVA_SPEC_VERSION >= 11 */
 
 /*[IF JAVA_SPEC_VERSION >= 20]*/
-	/* ToDo: Real implementation to be added under https://github.com/eclipse-openj9/openj9/issues/15933. */
+	/**
+	 * {@return an unmodifiable set of the {@linkplain AccessFlag access
+	 * flags} for this class, possibly empty}
+	 *
+	 * For an array class, the PUBLIC, PRIVATE and PROTECTED access flags should be the
+	 * same as those of its component type, and the FINAL access flag should always
+	 * present. For a primitive type or void, the PUBLIC and FINAL access flags should
+	 * always present, and the PROTECTED and PRIVATE access flags should always be absent.
+	 *
+	 * @since 20
+	 */
 	public Set<AccessFlag> accessFlags() {
-		return null;
+		int rawModifiers = getModifiersImpl();
+		/* Uses the implementation from getModifiers() and adds SUPER access flag to
+		 * the mask. We don't use getModifiers() directly and then add SUPER flag back
+		 * because the SUPER flag may or may not be set in the rawModifiers.
+		 */
+		if (isArray()) {
+			rawModifiers &= Modifier.PUBLIC | Modifier.PRIVATE | Modifier.PROTECTED |
+						Modifier.ABSTRACT | Modifier.FINAL;
```

Comment:
Per function header, 
> For an array class, the PUBLIC, PRIVATE and PROTECTED access flags should be the
same as those of its component type, and the FINAL access flag should always present.

Modifier.FINAL should always be part of this

Reply:
A note on this, the native impl of getModifiersImpl() forces ABSTRACT and FINAL to be added to the component type's modifiers, so forcing FINAL here may not be needed.